runtime/compiler/runtime/SymbolValidationManager.cpp

```diff
@@ -1286,6 +1286,16 @@ TR::SymbolValidationManager::validateStaticClassFromCPRecord(uint16_t classID, u
    {
    J9Class *beholder = getJ9ClassFromID(beholderID);
    J9ConstantPool *beholderCP = J9_CP_FROM_CLASS(beholder);
+
+   // This CP entry could still be unresolved, which would cause
+   // getClassOfStaticFromCP() to fail. Attempt to resolve it manually to reduce
+   // the chances of failure.
+   if (cpIndex != -1)
```

Comment:
Should we call `TR_ResolvedJ9Method::getClassOfStaticFromCP()` first and only if that fails try to resolve at compile time?
I am assuming that most of the time getClassOfStaticFromCP() will succeed so we don't pay the price of getting VM access.

Reply:
`TR_ResolvedJ9Method::getClassOfStaticFromCP` acquires vmaccess, so as the code stands, we'd be doing it twice. Three times if the approach is going to be:
```
if (!TR_ResolvedJ9Method::getClassOfStaticFromCP) {
  resolveStaticFieldRef;
  getClassOfStaticFromCP;
}
```
it would be nice if `J9::VMAccessCriticalSection` checked if we already have vmaccess and do an early return, though that would have the side effect of not checking if the compilation should be interrupted as often.

Reply:
Would it make sense to put this in `getClassOfStaticFromCP` instead?

Reply:
If the code already has VM access it won't acquire it again, so we can do
```
if (cp != -1)
   {
   TR::VMAccessCriticalSection getClassFromConstantPool(_fej9);
   TR_OpaqueClassBlock * clazz = TR_ResolvedJ9Method::getClassOfStaticFromCP(_fej9, beholderCP, cpIndex);
    if (!clazz)
       {
        _fej9->_vmFunctionTable->resolveStaticFieldRef(_fej9->vmThread(), NULL, beholderCP, cpIndex, J9_RESOLVE_FLAG_JIT_COMPILE_TIME, NULL);
        clazz = TR_ResolvedJ9Method::getClassOfStaticFromCP(_fej9, beholderCP, cpIndex);
       }
   } 
return validateSymbol(classID, clazz);
```

Reply:
> Would it make sense to put this in `getClassOfStaticFromCP` instead?

I think we might as well. I don't know of any reason why we shouldn't try to compile-time resolve if we call `getClassOfStaticFromCP()` from somewhere else